src/map/script.cpp

```diff
@@ -27493,6 +27493,31 @@ BUILDIN_FUNC(permission_add)
 	return SCRIPT_CMD_SUCCESS;
 }
 
+BUILDIN_FUNC(mesitemicon){
+	t_itemid nameid = script_getnum( st, 2 );
+	std::shared_ptr<item_data> data = item_db.find( nameid );
+	
+	if( data == nullptr ){
+		ShowError( "buildin_mesitemicon: Item ID %u does not exists.\n", nameid );
+		script_pushconststr( st, "" );
```

Comment:
Just a small thought, maybe we could use `ItemDatabase::create_item_icon_for_mes` that returns "Unknown item" when data is nullptr, like this
```suggestion
	const char* name = nullptr;

	if( script_hasdata( st, 3 ) ){
		name = script_getstr( st, 3 );
	}

	// Create the link, depending on configuration
	std::string itemlstr = item_db.create_item_icon_for_mes( data, name );

	if( data == nullptr ){
		ShowError( "buildin_mesitemicon: Item ID %u does not exists.\n", nameid );
		script_pushconststr( st, itemlstr );
```

Reply:
Is also okay for me.

Personally I would prefer SCRIPT_CMD_FAILURE and halt the script, but thats another chapter. ;-)

Reply:
Let's stick with your version